tests-cuda/test_1809_array_cuda_jit.py

```diff
@@ -0,0 +1,423 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import cupy as cp  # noqa: F401
+import numpy as np
+import pytest
+
+import awkward as ak
+
+try:
+    import numba as nb
+    import numba.cuda as nb_cuda
+    from numba import config
+
+    ak.numba.register_and_check()
+
+except ImportError:
+    nb = nb_cuda = None
+
+numbatest = pytest.mark.skipif(
+    nb is None or nb_cuda is None, reason="requires the numba and numba.cuda packages"
+)
+
+config.CUDA_LOW_OCCUPANCY_WARNINGS = False
+config.CUDA_WARN_ON_IMPLICIT_COPY = False
+
+
+@nb_cuda.jit(extensions=[ak.numba.array_view_arg_handler])
+def multiply(array, n, out):
+    tid = nb_cuda.grid(1)
+    out[tid] = array[tid] * n
+
+
+@nb_cuda.jit(extensions=[ak.numba.array_view_arg_handler])
```

Comment:
@jpivarski this explicit reference to `array_view_arg_handler` is how Numba executes our code (your earlier question)

Reply:
This is something _users_ have to invoke, right? When they write a

```python
@nb.cuda.jit
def function(...)
```

they'll have to explicitly add `extensions=[ak.numba.array_view_arg_handler]`, if I understand this correctly. In that case, we should name it something more memorable (if it can't be completely automatic, as it is in the case of CPU extensions).

What about

```python
@nb.cuda.jit(extensions=[ak.numba.cuda])
def function(...)
```

?

Reply:
> This is something users have to invoke, right? 

Yes. @ianna is tracking this in https://github.com/numba/numba/issues/8609, but right now it's explicit by the user.

```python
@nb.cuda.jit(extensions=[ak.numba.cuda])
def function(...)
```

I'd perhaps go even further
```python
@nb.cuda.jit(extensions=[ak.numba.cuda_ext])
def function(...)
```
Or `.cuda_extension`

Reply:
I'm watching that feature request, too, now.

As long as this explicit extension is needed, we want the name to be memorable enough that users don't need to look it up. Creating a JIT-compiled function is something one does at the start of a project or in a new Python prompt on the fly—when having to look up a name would cause real friction.